gnss_lib_py/parsers/precise_ephemerides.py

```diff
@@ -0,0 +1,811 @@
+"""Functions to process precise ephemerides .sp3 and .clk files.
+
+"""
+
+__authors__ = "Sriramya Bhamidipati"
+__date__ = "09 June 2022"
+
+import os
+import warnings
+
+from datetime import datetime, timedelta, timezone
+
+import numpy as np
+from scipy import interpolate
+
+from gnss_lib_py.parsers.ephemeris import EphemerisManager
+from gnss_lib_py.parsers.navdata import NavData
+from gnss_lib_py.utils.sim_gnss import find_sat
+from gnss_lib_py.utils.time_conversions import gps_millis_to_tow
+from gnss_lib_py.utils.time_conversions import datetime_to_gps_millis
+import gnss_lib_py.utils.constants as consts
+
+# Define the number of sats to create arrays for
+NUMSATS = {'gps': (32, 'G'),
+           'galileo': (36, 'E'),
+           'beidou': (46, 'C'),
+           'glonass': (24, 'R'),
+           'qzss': (3, 'J')}
+
+class Sp3:
+    """Class handling satellite position data from precise ephemerides
+
+    """
+    def __init__(self):
+        self.const = None
+        self.xpos = []
+        self.ypos = []
+        self.zpos = []
+        self.tym = []
+        self.utc_time = []
+
+    def __eq__(self, other):
+        """Checks if two Sp3() classes are equal to each other
+
+        Parameters
+        ----------
+        other : gnss_lib_py.parsers.precise_ephemerides.Sp3
+            Sp3 object that stores .sp3 parsed information
+
+        Returns
+        ----------
+        bool_check : bool
+            Flag (True/False) that indicates if Sp3 classes are equal
+        """
+        bool_check = (self.const == other.const) & \
```

Comment:
Cool function! If it doesn't break in your current tests, this can be a task for the future: in some cases, Python might throw a warning or error about being made to perform element-wise comparisons. For those cases I would recommend using `np.all` and `np.logical_and` to avoid any warnings.